doc/cygnus-ngsi/integration/orion_cygnus_spark.md

```diff
@@ -0,0 +1,216 @@
+...
+```
+
+[Top](#top)
+
+##<a name="section4"></a>Setting up spark Streaming
+Apache Spark will be installed from the URL resulting from entering at [Spark Download page](http://spark.apache.org/downloads.html) and selecting:
+
+* Spark release: 1.6.3
+* Package type: pre-built for Hadoop 2.3
+* Download type: direct download
+
+The URL the "download Spark" link points to must be copied and then downloaded in our machine (into our home dir, for instance) by doing:
+
+```
+$ cd ~
+$ wget http://d3kbcqa49mib13.cloudfront.net/spark-1.6.3-bin-hadoop2.3.tgz
+--2017-01-18 14:40:31--  http://d3kbcqa49mib13.cloudfront.net/spark-1.6.3-bin-hadoop2.3.tgz
+Resolviendo d3kbcqa49mib13.cloudfront.net (d3kbcqa49mib13.cloudfront.net)... 54.240.186.114, 54.240.186.149, 54.240.186.159, ...
+Conectando con d3kbcqa49mib13.cloudfront.net (d3kbcqa49mib13.cloudfront.net)[54.240.186.114]:80... conectado.
+Petición HTTP enviada, esperando respuesta... 200 OK
+Longitud: 273211089 (261M) [application/x-tar]
+Grabando a: “spark-1.6.3-bin-hadoop2.3.tgz”
+
+100%[===========================================================================================================================================>] 273.211.089 30,8MB/s   en 9,6s   
+
+2017-01-18 14:40:40 (27,3 MB/s) - “spark-1.6.3-bin-hadoop2.3.tgz” guardado [273211089/273211089]
+```
+
+The downloaded file must be uncompressed and the removed:
+
+```
+$ tar xzf spark-1.6.3-bin-hadoop2.3.tgz
+$ rm spark-1.6.3-bin-hadoop2.3.tgz
+```
+
+Now, Spark should be available under our home dir, thus let's change directory:
+
+```
+$ cd spark-1.6.3-bin-hadoop2.3
+```
+
+[Top](#top)
+
+##<a name="section5"></a>Testing everything together
+In order to test the whole architecture of our real time context information analysis we'll use an already developed analysis application Spark provides within its examples, called `JavaFlumeEventCount`:
+
+```
+$ ls examples/src/main/java/org/apache/spark/examples/streaming/ | \
+grep JavaFlumeEventCount.java
+JavaFlumeEventCount.java
+```
+
+`JavaFlumeEventCount` simply counts the number of Avro events received at the configured port. We'll run such an application by using Spark's `run-example` script:
+
+```
+$ ./bin/run-example streaming.JavaFlumeEventCount 0.0.0.0 5051
+Using Spark's default log4j profile: org/apache/spark/log4j-defaults.properties
+17/01/18 14:46:26 INFO StreamingExamples: Setting log level to [WARN] for streaming example. To override add a custom log4j.properties to the classpath.
+17/01/18 14:46:26 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
+-------------------------------------------
+Time: 1484750792000 ms
+-------------------------------------------
+Received 0 flume events.
+```
+
+As can be seen, no Avro events are read. Neverhteless, if we update some context entity among the subscribed ones, we'll see the application receives it:
```

Comment:
For the impacient ;), I'd warn about sometimes it takes some seconds from notification until "1 flume events" trace in the logs.